src/shortestpaths/bellman-ford.jl

```diff
@@ -37,31 +37,35 @@ function bellman_ford_shortest_paths(
     distmx::AbstractMatrix{T}=weights(graph)
     ) where T<:Real where U<:Integer
 
-    active = Set{U}(sources)
-    sizehint!(active, nv(graph))
+    active = falses(nv(graph))
+    for s in sources
+        active[s] = true
+    end
     dists = fill(typemax(T), nv(graph))
     parents = zeros(U, nv(graph))
     dists[sources] .= 0
     no_changes = false
+    new_active = falses(nv(graph))
 
     for i in one(U):nv(graph)
         no_changes = true
-        new_active = Set{U}()
-        for u in active
+        new_active.chunks .= 0
+        active_vertices = U[v for v in findall(active)]
```

Comment:
why not `U[a]` ?

Reply:
`active` is a BitArray

Reply:
so?
```
julia> a = falses(10); a[3:5] .= true;

julia> b = collect(11:20);

julia> b[a]
3-element Array{Int64,1}:
 13
 14
 15
```

Reply:
Oh, I see. `U` is a type. I thought it was an existing vector that you were indexing into.

In LG we don't construct vectors this way for this exact reason. Prefer `Vector{U}(findall(a))` instead.

Reply:
If I understand correctly from your example, do you mean `vertices[active]` instead of `U[active]` ?

Reply:
Oh okay I'm using convert instead, I will change to `Vector{U}(findall(a))`

Reply:
That's very nice, actually. Benchmarking shows that it's equivalent in speed to `findall`, so yeah - do that. It's easier to understand.

Reply:
`vertices[active]` or `Vector{U}(findall(a))` ?